gadgets/trace_tcpdrop/program.bpf.c

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// Based on tcpdrop(8) from BCC
+//
+// Copyright 2018 Netflix, Inc.
+// 30-May-2018    Brendan Gregg   Created this.
+// 15-Jun-2022    Rong Tao        Add tracepoint:skb:kfree_skb
+// Copyright 2023 Microsoft Corporation
+
+#include <vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+
+#include <gadget/buffer.h>
+#include <gadget/types.h>
+#include <gadget/macros.h>
+
+#define GADGET_TYPE_TRACING
+#include <gadget/sockets-map.h>
+
+#define TASK_COMM_LEN 16
+
+struct event {
+	struct gadget_l4endpoint_t src;
+	struct gadget_l4endpoint_t dst;
+
+	gadget_timestamp timestamp;
+	__u8 state;
+	__u8 tcpflags;
+	__u32 reason;
```

Comment:
```suggestion
	enum skb_drop_reason reason;
```

So the strings can be printed automatically.

Reply:
✅ Indeed better!
I have one observation here, not sure if it's something that can be tackled in any way. The reason strings may be trimmed due to screen space:
``` 
ubuntu@gifted-rattlesnake:~/go/src/github.com/inspektor-gadget/inspektor-gadget$ sudo -E ./ig run claudiafirstregistry.azurecr.io/gadget:claudia-convert-trace-tcpdrop-gadget
INFO[0000] Experimental features enabled                
RUNTIME.CONTAINERNAME                         SRC                      DST                      STATE                    TCPFLAGS                 REASON                   PID                      UID                      GID                     TASK            
netem                                         172.17.0.2:58545         1.1.1.1:20480            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            2                        2                        SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:58545         1.1.1.1:20480            4                        17                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            1                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:52963         1.1.1.1:47873            5                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:58545         1.1.1.1:20480            4                        17                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget            
netem                                         172.17.0.2:58545         1.1.1.1:20480            5                        16                       SKB_DROP_REASON_QDISC_D… 24188                    0                        0                       wget

Reply:
You can use `ellipsis: start` to avoid trimming the last part in this case.